### PR TITLE
add an lsp fast path test for swapping class order

### PR DESCRIPTION
**Pull request by froydnj**
### Motivation

This is pretty dumb, but it is the sort of thing that one might expect should take the fast path (even if it doesn't), so we might as well have it as a test.  It took me a little time to write the test and understand why, anyway.

### Test plan

See included automated tests.

### CODE REVIEW

File: test/testdata/lsp/fast_path/swap_class_order.1.rbupdate [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# typed: true
+# assert-slow-path: true
+
+module A
+  class C
+    def bar; end
```
Comment (jez) [inline test/testdata/lsp/fast_path/swap_class_order.1.rbupdate:6]:
```suggestion
    def method_on_c; end
```

File: test/testdata/lsp/fast_path/swap_class_order.1.rbupdate [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  end
+
+  class B
+    def foo; end
```
Comment (jez) [inline test/testdata/lsp/fast_path/swap_class_order.1.rbupdate:10]:
```suggestion
    def method_on_b; end
```

File: test/testdata/lsp/fast_path/swap_class_order.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# typed: true
+
+module A
+  class B
+    def foo; end
```
Comment (jez) [inline test/testdata/lsp/fast_path/swap_class_order.rb:5]:
```suggestion
    def method_on_b; end
```

File: test/testdata/lsp/fast_path/swap_class_order.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  end
+
+  class C
+    def bar; end
```
Comment (jez) [inline test/testdata/lsp/fast_path/swap_class_order.rb:9]:
```suggestion
    def method_on_c; end
```

**Review by jez — APPROVED** [review]
Suggested some different names because I had to do a double take at first.

Have you looked into why this is the case?

**Comment by jez** [comment]
I guess this is the reason?

https://github.com/sorbet/sorbet/blob/d315899bda4c76868181c98b748cc72a82ee0ef5/core/Symbols.cc#L2255-L2258

**Comment by froydnj** [comment]
> Have you looked into why this is the case?

I believe some of the debugging logic I put in for fields on the fast path indicated the the class hashes were different, which would point at the class hashing code you pointed out (ordering of the class members matters for hashing purposes).

**Pull request merged**